### PR TITLE
Add build script

### PR DESCRIPTION
**Pull request by wszdexdrf**
I believe this should be all that is needed. I don't think we can install OS-level dependencies. For udev rules, it is possible but then we would have to duplicate/include_str the `install_udev_rules()` function in `build.rs`

### CODE REVIEW

File: build.rs
Comment (afilini) [inline build.rs:4]: I would add the `--user` flag so that we only install it for the current user (so we don't need root privileges)

**Review by afilini — COMMENTED** [review]
I don't think using build scripts is the best way to do this: ideally we want to install hwi on the user's machine, not the builder's.

I think the best way is to expose a couple of functions, one to check if hwi is available (also checking its version) and one to install it. Then I would leave it up to the user of this library to call those functions whenever they want.

Most of the time I would expect hwi to be packaged together with the rust binaries in some ways: when using package managers like apt or rpm you could just add the python hwi as a dependency of your package. Alternatively when building a wallet into an AppImage you can probably bundle hwi in there as well.

File: src/interface.rs
Comment (danielabrozzoni) [inline src/interface.rs:376]: Can be changed to 
```
            Some(
                PyModule::import(py, "hwilib")
                    .ok()?
                    .getattr("__version__")
                    .expect("Should have a __version__")
                    .to_string(),
            )
```

Which is more readable 

I think it's fine to `expect` on `__version__`, what do you think?

File: src/interface.rs
Comment (danielabrozzoni) [inline src/interface.rs:365]: nit: Get **the** installed version of hwlib. Returns None if **hwi is not installed**.

**Comment by danielabrozzoni** [comment]
Once this is updated, I can ACK, merge and tag the new version (needed for https://github.com/bitcoindevkit/bdk/pull/682) - no rush tho :) We can totally start reviewing #682 before rust-hwi is published, you can just use rust-hwi master in the meantime

File: src/interface.rs
Comment (danielabrozzoni) [reply src/interface.rs]: It should be "Get **the** installed version"

**Comment by danielabrozzoni** [comment]
utACK 589c3cf7cb163a583f84018670df3e9dfb7d4a9a

**Review by danielabrozzoni — APPROVED** [review]

**Pull request merged**
